README.md

```diff
@@ -2,36 +2,219 @@
+- We use [Google AI platform](https://cloud.google.com/ai-platform/) for deploying docker images on GCP. Please make sure you have AI platform APIs enabled on your GCP project.
+- Please make sure `docker` is installed and running if you want to use local docker process for docker build, otherwise [create a cloud storage bucket](https://cloud.google.com/ai-platform/docs/getting-started-keras#create_a_bucket) for using [Google cloud build](https://cloud.google.com/cloud-build) for docker image build and publish.
+- Install [nbconvert](https://nbconvert.readthedocs.io/en/latest/) if you are using a notebook file as `entry_point` as shown in [usage guide #4](#detailed-usage-guide).
 
+### Install latest release
 
-### Install latest release:
-
-```
+```console
 pip install -U tensorflow-cloud
 ```
 
-### Install from source:
+### Install from source
 
-```
+```console
 git clone https://github.com/tensorflow/cloud.git
 cd cloud
 pip install .
 ```
 
-## Usage examples
+## High level overview 
+
+Tensorflow cloud package provides the `run` API for training your models on GCP. Before we get into the details of the API, let's see how a simple workflow will look like using this API.
+
+1. Let's say you have a Keras model training code, such as the following, saved as `mnist_example.py`.
+
+```python
+import tensorflow as tf
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
+
+mnist_train = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+mnist_test = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+
+BUFFER_SIZE = 10000
+BATCH_SIZE = 64
+
+def scale(image, label):
+    image = tf.cast(image, tf.float32)
+    image /= 255
+    return image, label
+
+train_dataset = mnist_train.map(scale).cache().shuffle(
+    BUFFER_SIZE).batch(BATCH_SIZE)
+eval_dataset = mnist_test.map(scale).batch(BATCH_SIZE)
+
+model = tf.keras.Sequential([
+      tf.keras.layers.Flatten(input_shape=(28, 28)),
+      tf.keras.layers.Dense(512, activation='relu'),
+      tf.keras.layers.Dropout(0.2),
+      tf.keras.layers.Dense(10, activation='softmax')
+    ])
+
+model.compile(loss='sparse_categorical_crossentropy',
+              optimizer=tf.keras.optimizers.Adam(),
+              metrics=['accuracy'])
+
+model.fit(train_dataset, epochs=12)
+```
+
+2. After you have tested this model on your local environment for a few epochs, probably with a small dataset, you can train the model on Google cloud by writing the following simple script `scale_mnist.py`.
+
+```python
+import tensorflow_cloud as tfc
+tfc.run(entry_point='mnist_example.py')
+```
+
+Running this script will automatically apply Tensorflow [Mirrored distribution strategy](https://www.tensorflow.org/api_docs/python/tf/distribute/MirroredStrategy) and train your model at scale on Google Cloud Platform. Please see the [usage guide](#usage-guide) section for detailed instructions on how to use the API.
```

Comment:
Missing 'detailed' (or else remove it from the title below)

Reply:
Good catch. Removed detailed.